consensus/reactor.go

```diff
@@ -629,6 +629,7 @@ OUTER_LOOP:
 			// Load the block commit for prs.Height,
 			// which contains precommit signatures for prs.Height.
 			commit := conR.conS.blockStore.LoadBlockCommit(prs.Height)
+			commit.SetAddresses(conR.conS.Validators.GetAddresses())
```

Comment:
I don't think this is the correct validator set - it's using our current validator set but the commit is coming from the peers height, which is less than ours in this `if` clause.

The ease of getting something like this wrong is a bit uncomfortable. I wonder if there's still a better way - maybe trying to stuff both vote and commit into the VoteSetReader isn't the best design?

Reply:
Maybe instead we can define a method that takes a blockStore and a state and loads the full vote set (ie. the commits plus the addresses)

Reply:
yeah this is the wrong validatorset... we're fetching the commit for height prs.Height (a peer's height) and setting the current local validatorset...

Reply:
Yup, good catch @ebuchman. Are we currently storing the historical validator sets anywhere? Because it not we'll have to start doing that to be able to reconstruct votes with addresses for arbitrary heights like this.

Reply:
Yes you can get them from the state! Let's make a function like `GetVoteSetFromCommit(blockStore, state) VoteSet`

Reply:
Mmk that makes sense, I can work that up.

Reply:
Could we include a Commit.Addresses ([]address) and Commit.IsNils (bitarray) that are in the first iteration of this software mandatory to include, and we can work on making them optional after launch?  This way we won't introduce any bugs like the above, at least until we start working on optimizing Commit further, but I'm not sure we even need to do that any time soon.  20*100 = 2KB, compared to the signatures it's not that much more, and it helps with debugging commits, so we'll want it for the time being anyways.